src/Components/ColorsGallery/ColorsGallery.tsx

```diff
@@ -1,94 +1,78 @@
 import * as React from "react";
 import { StyleSheet, View, Text, ScrollView } from "react-native";
-import Colors, { colorTheme } from "src/Themes/Colors";
-import LinearGradient from "src/Components/LinearGradient/LinearGradient";
+import Colors from "src/Themes/Colors";
 import { isWeb } from "src/Helpers/platform";
-type Props = {
-  theme: "light" | "dark";
-};
+import withTheme, { ThemeProps, ThemeParamsType } from "src/Themes/withTheme";
+
+type Styles = ReturnType<typeof getStyles>;
+interface Props extends ThemeProps<Styles> {}
 
 class ColorsGallery extends React.PureComponent<Props> {
-  static defaultProps = {
-    theme: "light",
-  };
   public render() {
     return (
-      <ScrollView style={styles.container}>
-        <View style={styles.colorViewContainer}>
+      <ScrollView style={this.props.styles.container}>
+        <View style={this.props.styles.colorViewContainer}>
           {this.renderColors()}
-          <Text style={[styles.text, { fontSize: 18 }]}>Palette Colors</Text>
+          <Text style={[this.props.styles.text, { fontSize: 18 }]}>Palette Colors</Text>
           {this.renderPaletteColors()}
         </View>
       </ScrollView>
     );
   }
 
   renderPaletteColors = () => {
-    const theme = colorTheme[this.props.theme].palette;
+    const theme = this.props.colors.palette;
     const colorNames = Object.keys(Colors.palette);
     return colorNames.map((colorName) => {
       return (
         <View>
-          <Text style={styles.text}>{colorName}</Text>
-          <View style={[styles.colorView, { backgroundColor: theme[colorName] }]} />
+          <Text style={this.props.styles.text}>{colorName}</Text>
+          <View style={[this.props.styles.colorView, { backgroundColor: theme[colorName] }]} />
         </View>
       );
     });
   };
 
   renderColors = () => {
-    const theme = colorTheme[this.props.theme];
-    const colorNames = Object.keys(Colors);
-    return colorNames.map((colorName) => {
-      if (colorName !== "palette") {
-        if (typeof theme[colorName] === "string") {
-          return (
-            <View>
-              <Text style={styles.text}>{colorName}</Text>
-              <View style={[styles.colorView, { backgroundColor: theme[colorName] }]} />
-            </View>
-          );
-        }
-        return (
-          <View>
-            <Text style={styles.text}>{colorName}</Text>
-            <LinearGradient
-              style={styles.colorView}
-              gradientStartColor={theme[colorName].gradientColor1}
-              gradientEndColor={theme[colorName].gradientColor2}
-            />
-          </View>
-        );
-      }
-      return <View />;
+    const colorNames = Object.keys(this.props.colors);
+    return colorNames.map((color) => {
+      if (color === "palette" || typeof this.props.colors[color] !== "string") return <View />;
```

Comment:
Why do we need to check `typeof` value of color object, if we have typescript to do this job for us?

Reply:
`However, having to define a function to figure out if a type is a primitive is kind of a pain. Luckily, you don’t need to abstract typeof x === "number" into its own function because TypeScript will recognize it as a type guard on its own. That means we could just write these checks inline.`